filebeat/input/filestream/input_integration_test.go

```diff
@@ -100,3 +147,116 @@ func TestFilestreamCloseEOF(t *testing.T) {
 
 	env.requireOffsetInRegistry(testlogName, expectedOffset)
 }
+
+// test_empty_lines from test_harvester.py
+func TestFilestreamEmptyLine(t *testing.T) {
+	env := newInputTestingEnvironment(t)
+
+	testlogName := "test.log"
+	inp := env.mustCreateInput(map[string]interface{}{
+		"paths":                             []string{env.abspath(testlogName)},
+		"prospector.scanner.check_interval": "1ms",
+	})
+
+	ctx, cancelInput := context.WithCancel(context.Background())
+	env.startInput(ctx, inp)
+
+	testlines := []byte("first log line\nnext is an empty line\n")
+	env.mustWriteLinesToFile(testlogName, testlines)
+
+	env.waitUntilEventCount(2)
+	env.requireOffsetInRegistry(testlogName, len(testlines))
+
+	moreTestlines := []byte("\nafter an empty line\n")
+	env.mustAppendLinesToFile(testlogName, moreTestlines)
+
+	env.waitUntilEventCount(3)
```

Comment:
If I understand the test correctly we should drop the empty line. Does `waitUntilEventCount` return if the event count is 4? Maybe we should rather validate the actual event contents here.

Reply:
Well, it tries to wait until 3 events arrive and after a while it timeouts. It could fail earlier if we have too many events to speed up things.

Reply:
It is not about failing early. I wonder how we can be sure the empty line is dropped as expected if we do not validate the actual events contents that has been published.